general/g.proj/output.c

```diff
@@ -21,64 +21,99 @@
 #include <grass/gprojects.h>
 #include <grass/glocale.h>
 #include <grass/config.h>
+#include <grass/parson.h>
 
 #ifdef HAVE_OGR
 #include <cpl_csv.h>
 #endif
 
 #include "local_proto.h"
 
-static int check_xy(int shell);
+static int check_xy(enum OutputFormat);
+static void print_json(JSON_Value *);
 
 /* print projection information gathered from one of the possible inputs
  * in GRASS format */
-void print_projinfo(int shell)
+void print_projinfo(enum OutputFormat format)
 {
     int i;
+    JSON_Value *value;
+    JSON_Object *object;
 
-    if (check_xy(shell))
+    if (check_xy(format))
         return;
 
-    if (!shell)
+    if (format == PLAIN)
         fprintf(
             stdout,
             "-PROJ_INFO-------------------------------------------------\n");
+
+    if (format == JSON) {
+        value = json_value_init_object();
+        object = json_object(value);
+    }
+
     for (i = 0; i < projinfo->nitems; i++) {
         if (strcmp(projinfo->key[i], "init") == 0)
             continue;
-        if (shell)
+        switch (format) {
+        case PLAIN:
             fprintf(stdout, "%s=%s\n", projinfo->key[i], projinfo->value[i]);
-        else
+            break;
+        case SHELL:
             fprintf(stdout, "%-11s: %s\n", projinfo->key[i],
                     projinfo->value[i]);
+            break;
+        case JSON:
+            json_object_set_string(object, projinfo->key[i],
+                                   projinfo->value[i]);
+            break;
+        }
     }
 
     /* TODO: use projsrid instead */
     if (projsrid) {
-
-        if (!shell) {
+        switch (format) {
+        case PLAIN:
             fprintf(stdout, "-PROJ_SRID----------------------------------------"
                             "---------\n");
             fprintf(stdout, "%-11s: %s\n", "SRID", projsrid);
-        }
-        else
+            break;
+        case SHELL:
             fprintf(stdout, "%s=%s\n", "srid", projsrid);
+            break;
+        case JSON:
+            json_object_set_string(object, "srid", projsrid);
+            break;
+        }
     }
 
     if (projunits) {
-        if (!shell)
+        if (format == PLAIN)
             fprintf(stdout, "-PROJ_UNITS---------------------------------------"
                             "---------\n");
         for (i = 0; i < projunits->nitems; i++) {
-            if (shell)
-                fprintf(stdout, "%s=%s\n", projunits->key[i],
-                        projunits->value[i]);
-            else
+            switch (format) {
+            case PLAIN:
                 fprintf(stdout, "%-11s: %s\n", projunits->key[i],
                         projunits->value[i]);
+                break;
+            case SHELL:
+                fprintf(stdout, "%s=%s\n", projunits->key[i],
+                        projunits->value[i]);
+                break;
+            case JSON:
+                json_object_set_string(object, projunits->key[i],
```

Comment:
Is there a way to know if the current projunit is or should be represented as a number?

Reply:
The `unit` and `units` key values will always be a string. No matter the unit (km, ft, degree, etc..) the third key (`meters`) in projunits is always converted to meters. You can see the conversions on line 43 in `lib/proj/convert.c`